File: Domain/Activity.cs
namespace Domain;

public sealed class Activity
{
    public Guid Id { get; set; }
    public string Title { get; set; } = default!;
    public DateTime Date { get; set; }
    public string Description { get; set; } = default!;
    public string Category { get; set; } = default!;
    public string City { get; set; } = default!;
    public string Venue { get; set; } = default!;
    public bool IsCancelled { get; set; }
    public ICollection<ActivityAttendee> Attendees { get; set; } = new List<ActivityAttendee>();
    public ICollection<Comment> Comments { get; set; } = new List<Comment>();
}

File: Web/src/models/activity.ts
export interface Activity {
  id: string;
  title: string;
  date: Date;
  description: string;
  category: string;
  city: string;
  venue: string;
  hostUsername: string;
  isCancelled: boolean;
  isGoing: boolean;
  isHost: boolean;
  host: ActivityAttendee;
  attendees: ActivityAttendee[];
}

export type ActivityFormValues = Pick<
  Activity,
  "id" | "title" | "description" | "category" | "date" | "city" | "venue"
>;

export interface ActivityAttendee {
  username: string;
  displayName: string;
  image?: string;
  bio?: string;
}
